File: server/core-finance/src/main/java/com/arcabank/core_finance/dto/AccountResponse.java
package com.arcabank.core_finance.dto;

import lombok.Builder;

import java.math.BigDecimal;
import java.util.UUID;

@Builder
public record AccountResponse(
    UUID accountId,
    UUID cardId,
    String iban,
    String cardNumber,
    String cardHolderName,
    String expirationDate,
    String cvv,
    String currency,
    BigDecimal balance
) {
}

File: client/src/app/response/AccountResponse.ts
export interface CreateAccountResponse {
 id: string;
 userId: string;
 iban?: string;
 currency: string;
 balance: number;
 status: string;
 createdAt: string;
}



export interface CreateAccountWithCardResponse {
  accountId: string;
  cardId: string;
  iban: string;
  cardNumber: string;
  cardHolderName: string;
  expirationDate: string;
  cvv: string;
  currency: string;
  balance: number;
}


export interface accountBlockUnblockMessageResponse {
  message: string;
}

export interface Transaction {
  id: string;
  senderAccountId: string;
  receiverAccountId: string;
  amount: number;
  currency: string;
  status: string;
  createdAt: string;
  type: string;
}

export interface getAccountTransactionsResponse {
  content: Transaction[];

  pageNumber: number;
  pageSize: number;

  totalElements: number;
  totalPages: number;
}
